include/fmt/core.h

```diff
@@ -2656,21 +2656,28 @@ constexpr auto get_arg_index_by_name(basic_string_view<Char> name) -> int {
   if constexpr (detail::is_statically_named_arg<T>()) {
     if (name == T::name) return N;
   }
-  if constexpr (sizeof...(Args) > 0)
+  if constexpr (sizeof...(Args) > 0) {
     return get_arg_index_by_name<N + 1, Args...>(name);
-  (void)name;  // Workaround an MSVC bug about "unused" parameter.
-  return invalid_arg_index;
+  } else {
+    (void)name;  // Workaround an MSVC bug about "unused" parameter.
+    return invalid_arg_index;
+  }
 }
 #endif
 
 template <typename... Args, typename Char>
 FMT_CONSTEXPR auto get_arg_index_by_name(basic_string_view<Char> name) -> int {
 #if FMT_USE_NONTYPE_TEMPLATE_PARAMETERS
-  if constexpr (sizeof...(Args) > 0)
+  if constexpr (sizeof...(Args) > 0) {
```

Comment:
Looks like there will be an unbalanced `{` if `FMT_USE_NONTYPE_TEMPLATE_PARAMETERS` is defined.

Reply:
No. There are fully balanced `if`-`else` branches. Otherwise the code wouldn't compile on my system, like at all. I've checked with `FMT_USE_NONTYPE_TEMPLATE_PARAMETERS` overridden to both `0`and `1`, and with `1` being auto-detected.

Reply:
My bad, it does look correct.

Reply:
@DanielaE: While you on it, please be aware of this: 
https://developercommunity.visualstudio.com/t/In-MSVC-16101-Intellisense-and-compil/1449427

Reply:
It's not uncommon for EDG (the intellisense compiler) be out of sync with the actual compiler during the preview phase. As long as msvc reports correctly (and it does), we're fine.